### PR TITLE
more work for PHP 8

### PR DESCRIPTION
**Pull request by remicollet**
Mostly about property function arg change from zval to zend_object.

### CODE REVIEW

**Comment by remicollet** [comment]
Using PHP_VERSION : 8.0.0beta2

```
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   16
---------------------------------------------------------------------

Number of tests :  175               166
Tests skipped   :    9 (  5.1%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :    0 (  0.0%) (  0.0%)
Tests passed    :  166 ( 94.9%) (100.0%)
---------------------------------------------------------------------
Time taken      :   80 seconds
=====================================================================

```

File: amqp_channel.c [hunk 1]
```diff
@@ -331,7 +336,7 @@ static PHP_METHOD(amqp_channel_class, __construct)
 	amqp_connection_object *connection;
 
 	/* Parse out the method parameters */
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "o", &connection_object) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "O", &connection_object, amqp_connection_class_entry) == FAILURE) {
```
Comment (remicollet) [inline amqp_channel.c:339]: Ensure parameter type is checked (warning with php 5 and 7, TypeError exception with php 8)

File: amqp_exchange.c [hunk 2]
```diff
@@ -65,20 +65,20 @@ static PHP_METHOD(amqp_exchange_class, __construct)
 	zval *channelObj;
 	amqp_channel_resource *channel_resource;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "o", &channelObj) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "O", &channelObj, amqp_channel_class_entry) == FAILURE) {
```
Comment (remicollet) [inline amqp_exchange.c:68]: Ensure parameter type is checked...

File: amqp_queue.c [hunk 3]
```diff
@@ -68,20 +68,20 @@ static PHP_METHOD(amqp_queue_class, __construct)
 	zval *channelObj;
 	amqp_channel_resource *channel_resource;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "o", &channelObj) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "O", &channelObj, amqp_channel_class_entry) == FAILURE) {
```
Comment (remicollet) [inline amqp_queue.c:71]: Ensure parameter type is checked...

**Comment by remicollet** [comment]
For constructor, for consistency, perhaps better to switch to zend_parse_parameters_throw

**Comment by lstrojny** [comment]
Looks good, thank you!

**Comment by Jan-E** [comment]
win32/php_stdint.h does not exist anymore in PHP 8. A generic solution could use something like this
```
#if defined(_MSC_VER) && _MSC_VER <= 1916
#include <win32/php_stdint.h>
#else
#include <main/php_stdint.h>
#endif
```
It checks if a MSVC compiler is used and if the VC compiler is VC15 or less.

**Pull request merged**